### DISCUSSION TITLE
Shear rate fluctuation and calculating the viscous stress tensor

### DISCUSSION THREAD

**Discussion by ahmedelhanafy2022** [UvaCsl/HemoCell#74] [General discussion]
Hello all,
I have read two published papers that are based on HemoCell and I have two inquiries about the obtained results:

The first paper: **Haemodynamic flow conditions at the initiation of high-shear platelet aggregation: a combined in vitro and cellular in silico study**

I found that there are high fluctuations in the shear rate obtained in Fig.9. Is there an explanation for this behavior?

The second paper:**The effect of stiffened diabetic red blood cells on wall shear stress in a reconstructed 3D microaneurysm**

Regarding Eq. 1 for calculating the viscous stress tensor,  where can I find it in HemoCell output ?
Do you have any idea about that?
Regards 
Ahmed

**Comment by gzavo** [comment]
Hi @ahmedelhanafy2022,

> I found that there are high fluctuations in the shear rate obtained in Fig.9. Is there an explanation for this behavior?

I assume you mean the light blue and light red curves. The fluctuation is due to the cellular nature. When a cell passes close to the wall the shear stress imposed on the wall is increased.

> Regarding Eq. 1 for calculating the viscous stress tensor,  where can I find it in HemoCell output ?

Look at some of the relevant tensors quantities you can output by adding them to the ‘hemocell.setFluidOutputs’ function arguments:
https://github.com/UvaCsl/HemoCell/blob/3189621a2118152addefd3a6f5852536c3fd2d14/io/FluidHdf5IO.hh#L184-L198

Computing it manually from the velocity field is also a (more laborious) option.

**Comment by ahmedelhanafy2022** [comment]
That's great!
Many thanks Dr. @gzavo.
It seems to be a trivial question, could you tell me how can I add them? 
I tried to find it in **HemoCell/core/hemoCell.cpp** but I could not

**Reply by ahmedelhanafy2022** [reply]
Finally, I got it!
It can be found in cpp file of each case.
Many thanks @gzavo
